compiler_gym/envs/llvm/service/LlvmEnvironment.cc

```diff
@@ -340,74 +340,74 @@ Status LlvmEnvironment::getObservation(LlvmObservationSpace space, Observation*
     case LlvmObservationSpace::IR_INSTRUCTION_COUNT: {
       const auto cost =
           getCost(LlvmCostFunction::IR_INSTRUCTION_COUNT, benchmark().module(), workingDirectory_);
-      reply->mutable_int64_list()->add_value(static_cast<int64_t>(cost));
+      reply->set_scalar_int64(static_cast<int64_t>(cost));
```

Comment:
I like that! We don't even need to be ad-hoc, we can use `setattr()` to create new methods on the observation class on-the-fly so that every observation space will automatically get its own getter:

```py
>>> env = gym.make("llvm-v0")
>>> env.reset()
>>> env.observation.Autophase()
array([  0,   0,  46,  40,   3,  45,  16,   2,  14,  18,   1,   0,   0,
        67,   0,  63,   5,   4,  81,  76,  38,  51,  52,  45,  15,   0,
        14,  37,   0,   8,  59,   6,  63,  16,  35,  18,   0, 117,   0,
         0,   0,   4,  28,   0,   0,  61,   0,   0,   0,   0,  67, 414,
       266,  10,   0, 195])
```

This seems related to #74, I'll implement it and push there

Reply:
I like that! 
Does it work for IDEs?

Reply:
No :( 7da88ffa26dbc2be321089e9413ef792444d4a98

but can work in conjunction with the generated enums

Reply:
How about this: The generic one works for any new ones + hard code the known ones?
Or just the generic :-)